backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/VmManagementCommandBase.java

```diff
@@ -116,15 +119,22 @@ protected boolean setAndValidateCpuProfile() {
     }
 
     protected boolean validateCPUHotplug(VmStatic vmStaticData) {
-        if (getVm().isRunningOrPaused()) {
-            // Can't set more CPUs than available on the host where VM is running.
-            // Potential overcommit (interference with other running VMs) will be resolved by scheduler.
-            // In alignment with the CPUPolicyUnit, VM's hyperthreading is not considered.
-            if (getVds() != null && vmStaticData.getNumOfCpus(false) > getVds().getCpuThreads()) {
-                return false;
-            }
+        // Can't set more CPUs than available on the host where VM is running.
+        // Potential overcommit (interference with other running VMs) will be resolved by scheduler.
+        // In alignment with the CPUPolicyUnit, VM's hyperthreading is not considered.
+        int availableThreads = getVds().getCpuThreads() - vdsCpuUnitPinningHelper.countUnavailableCpus(
+                getVdsManager(getVdsId()).getCpuTopology(), true);
```

Comment:
don't we mix cores and threads again? getVds().getCpuThreads() returns the amount of threads on the host and vdsCpuUnitPinningHelper#countUnavailableCpus return the amount of taken cores (with the default configuration of countThreadsAsCores=false)

Reply:
This part was changed now with alignment to the CPUPolicyUnit.

Reply:
looks good now